src/runtime-pmix.c

```diff
@@ -131,6 +158,45 @@ shmem_runtime_exchange(void)
     bool wantit=true;
     //bool active = true;
 
+    if (node_ranks) {
```

Comment:
Can we populate the node info during init instead of delaying it to the exchange step?

Reply:
I believe we can do this anytime after `PMIx_Init`, but I wasn't sure if we wanted to keep the organization consistent across PMI runtimes.  I'm happy either way.